fiftyone/core/utils.py

```diff
@@ -1992,3 +1995,26 @@ def validate_hex_color(value):
 
 
 fos = lazy_import("fiftyone.core.storage")
+
+
+def load_dataset(identifier):
+    """Loads the dataset from the database with the given name or id."""
```

Comment:
I propose updating the signature of this method to:

```py
def load_dataset(id=None, name=None):
```

where the user can provide either the `name` or `id` of the dataset as a kwarg.

Although it is an edge case, the current implementation does not allow for the possibility that the user assigns a dataset name like `name=str(ObjectId())`, as it would be assumed that is the `_id`, not the `name` of the dataset.

As for the implementation of the method, since you cannot directly load the `DatasetDocument` outside of the `Dataset` constructor, if the user provides an `id`, you'll just have to lookup the `name` and then load the dataset using the name in a second step. It's unfortunate that it requires two database connections, but it would take more effort to avoid that given the migration requirements and also the fact that `Dataset` objects are singletons keyed by `name`, not `id`.

Reply:
How does it take more effort? The current implementation accomplishes this.

Reply:
Again, like you said above, this is NOT user facing. It can be used for debugging or internally when you need to access the dataset in some intermediate call.

Reply:
Also loading the dataset via id uses pymongo directly to get the dataset doc from the db, but to just read the name from that query to then load the full dataset via mongoengine doc.objects, you are then attempting to load a cached dataset doc from the mongoengine cache, which may be causing errors like "dataset-name doesn't exist" even though you just created it or creating corrupted datasets/missing data when mixing local sdk usage with UI/in app changes.

For ref:
```
As of MongoEngine 0.8 the querysets utilise a local cache. So iterating it multiple times will only cause a single query.
```
https://docs.mongoengine.org/guide/querying.html